readthedocs/oauth/services/github.py

```diff
@@ -100,37 +100,26 @@ def create_repository(self, fields, privacy=None, organization=None):
         ]):
 
             repo, _ = RemoteRepository.objects.get_or_create(
-                remote_id=fields['id'],
-                vcs_provider=self.vcs_provider_slug
-            )
-            remote_repository_relation = repo.get_remote_repository_relation(
-                self.user, self.account
+                remote_id=str(fields["id"]),
+                vcs_provider=self.vcs_provider_slug,
             )
 
-            # It's possible that a user has access to a repository from within
-            # an organization without being member of that organization
-            # (external contributor). In this case, the repository will be
-            # listed under the ``/repos`` endpoint but not under ``/orgs``
-            # endpoint. Then, when calling this method (``create_repository``)
-            # we will have ``organization=None`` but we don't have to skip the
-            # creation of the ``RemoteRepositoryRelation``.
-            if repo.organization and organization and repo.organization != organization:
-                log.debug(
-                    'Not importing repository because mismatched orgs.',
-                    repository=fields['name'],
-                )
-                return None
-
-            if any([
-                # There is an organization associated with this repository:
-                # attach the organization to the repository
-                organization is not None,
-                # There is no organization and the repository belongs to a
-                # user: removes the organization linked to the repository
-                not organization and fields['owner']['type'] == 'User',
-            ]):
+            owner_type = fields["owner"]["type"]
+
+            # If there is an organization associated with this repository,
+            # attach the organization to the repository.
+            if (
+                organization
+                and owner_type == "Organization"
+                and organization.remote_id == str(fields["owner"]["id"])
+            ):
```

Comment:
that could work too, but when the organization isn't `None` it's guaranteed that we already created the organization, and we would be querying the org n times instead of 1.

Reply:
I think it's better to do `get_or_create` as I mentioned in the other comment. It simplifies the code and make it clear, even if we have to do N queries. They will be pretty quick. We could also make a `cached_property` or similar if we want to optimize it, but I don't think it is worth that complexity.

I prefer this piece of code to be as clean/simple as possible since it's one of the key parts of our SSO and we have had already enough problems that are hard to track/debug.